sdk-base/src/main/java/com/mapbox/maps/plugin/animation/MapboxAnimatorThread.kt

```diff
@@ -0,0 +1,184 @@
+  constructor() : this(AndroidManifestMetaData())
+
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal constructor(manifestMetaData: AndroidManifestMetaData) {
+    this.manifestMetaData = manifestMetaData
+  }
+
+  private val manifestMetaData: AndroidManifestMetaData
+  private val mainAnimatorThread: MainAnimatorThread = MainAnimatorThread()
+  private var backgroundAnimatorThread: BackgroundAnimatorThread? = null
+
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal val backgroundLooper: Looper?
+    get() = backgroundAnimatorThread?.handler?.looper
+
+  /**
+   * The beginning of the animator thread lifecycle. This will create resources for running
+   * animator threads.
+   */
+  fun onAttached() {
+    mainAnimatorThread.onAttached()
+    if (manifestMetaData.backgroundAnimatorThreadEnabled()) {
+      backgroundAnimatorThread = BackgroundAnimatorThread().onAttached()
+    }
+  }
+
+  /**
+   * Allows you to post any job onto the animator thread. This can be used to trigger animations
+   * that may not be using the [Animator].
+   *
+   * Run immediate if my looper is the designated looper, otherwise post the work to a handler
+   * queue. This is to reduce thread context switches while running animations.
+   */
+  fun post(runnable: Runnable): Boolean {
+    return (backgroundAnimatorThread ?: mainAnimatorThread).post(runnable).also { posted ->
+      if (!posted) {
+        logW(
+          TAG,
+          "post did not happen, ensure the app is in foreground. background  animator " +
+            "thread: ${backgroundAnimatorThread != null}"
+        )
+      }
+    }
+  }
+
+  /**
+   * Mapbox gl-native libraries require the main thread. Use this function whenever an animation
+   * task should be posted on the main thread, even when background threads are enabled.
+   *
+   * Run immediate if my looper is the designated looper, otherwise post the work to a handler
+   * queue. This is to reduce thread context switches while running animations.
+   */
+  fun postMain(runnable: Runnable): Boolean {
+    return mainAnimatorThread.post(runnable).also { posted ->
+      if (!posted) {
+        logW(TAG, "postMain did not happen, ensure the app is in foreground")
+      }
+    }
+  }
+
+  /**
+   * The beginning of the animator thread lifecycle. This will create resources for running
+   * animator threads.
+   */
+  fun onDetached() {
+    mainAnimatorThread.onDetached()
+    backgroundAnimatorThread?.onDetached()
+    backgroundAnimatorThread = null
+  }
+
+  private companion object {
+    private const val TAG = "MapboxAnimatorThread"
+  }
+}
+
+/**
+ * The [PackageManager] has issues being mocked. This class makes it so usages of the package
+ * meta data can be tested without mocking the application info metadata.
+ *
+ * https://github.com/mockk/mockk/issues/182
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+internal class AndroidManifestMetaData {
```

Comment:
as described in another comment - I would not control if background thread is needed thru manifest but would make it runtime API for AA case only

Reply:
> runtime API for AA case only

All the issues that are exposed when this is enabled, will happen for the AA case alone. The AA case is completely messed up - so I'm using this to discover why it is so messed up.